### ISSUE TITLE
RKE1 and RKE2 Clusters are stuck in Removing state

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#35180]
**Rancher Server Setup**
- Rancher version: v2.6.1
- Installation option (Docker install/Helm Chart): docker install
- Proxy/Cert Details: rancher generated

**Information about the Cluster**
- Kubernetes version: `v1.21.5+rke2r2` and `1.21.5`
- Cluster Type (Local/Downstream): RKE2 DO

**Describe the bug**
RKE1 and RKE2 Clusters are stuck in Removing state

**To Reproduce**
- Deployed 2 rke2 DO cluster
- **Disable RKE2 feature in rancher**
- Deploy an RKE1 cluster
- From Cluster Management, delete the clusters
- All the clusters are stuck in removing state 
- RKE2 nodes are not deleted. Logged separate bug - https://github.com/rancher/rancher/issues/35179
- Rancher logs show these logs repeatedly:
```
2021/10/19 15:46:45 [ERROR] error syncing 'c-m-8wdhwsft': handler mgmt-cluster-remove: Index with name by-cluster does not exist, requeuing
2021/10/19 15:46:54 [ERROR] error syncing 'c-kjlxd': handler mgmt-cluster-remove: Index with name by-cluster does not exist, requeuing
```

**Scenario 2**

- Deploy an RKE1 downstream DO cluster
- Disable RKE2 feature flag
- Deploy another RKE1 downstream cluster DO
- Select both the clusters and hit Delete
- The clusters are stuck in removing state
- Rancher logs:
```
2021/10/20 18:00:27 [ERROR] error syncing 'c-6xzkd': handler cluster-watch: namespaces "c-6xzkd" not found, handler mgmt-cluster-remove: Index with name by-cluster does not exist, requeuing
2021/10/20 18:01:39 [ERROR] Error during subscribe websocket: close sent
2021/10/20 18:01:39 [ERROR] Error during subscribe websocket: close sent
2021/10/20 18:01:40 [ERROR] error syncing 'c-2mkz4': handler mgmt-cluster-remove: Index with name by-cluster does not exist, requeuing
2021/10/20 18:01:56 [ERROR] error syncing 'c-6xzkd': handler mgmt-cluster-remove: Index with name by-cluster does not exist, requeuing
2021/10/20 18:03:40 [ERROR] error syncing 'c-2mkz4': handler mgmt-cluster-remove: Index with name by-cluster does not exist, requeuing
2021/10/20 18:03:56 [ERROR] error syncing 'c-6xzkd': handler mgmt-cluster-remove: Index with name by-cluster does not exist, requeuing
```
- The nodes are deleted from the backend

As per @thedadams the following usecases will be affected
```

- If you provision a cluster with RKE2 off then turn it on and then back off, then cluster will be affected.
- If you provision a cluster with RKE2 on then turn it off, then the cluster will be affected.
- If you provision a cluster with RKE2 off and never turn it on, then the cluster should be fine.

```

**Comment by sowmyav27** [comment]
As per @thedadams 
```
It was decided that a user cannot turn RKE2 provisioning off if there are any provisioned RKE2/K3S. If there are only RKE1/KEv2/imported clusters, then you can turn off RKE2 feature flag.
```

**Comment by ronhorton** [comment]
<img width="989" alt="Screen Shot 2021-11-11 at 1 18 17 PM" src="https://user-images.githubusercontent.com/89212856/141349476-6218b044-931a-42a6-9c20-26b93b039938.png">

The RKE1 Clusters that I provisioned were deleted, but my rke2 cluster will not delete. Seeing this in docker logs:
`E1111 18:19:35.835226      35 reflector.go:139] pkg/mod/github.com/rancher/client-go@v0.21.0-rancher.1/tools/cache/reflector.go:168: Failed to watch *v1.Node: failed to list *v1.Node: {"Code":{"Code":"Forbidden","Status":403},"Message":"clusters.management.cattle.io \"c-m-47jtl6kl\" is forbidden: User \"u-s6jkdanb7v\" cannot get resource \"clusters\" in API group \"management.cattle.io\" at the cluster scope","Cause":null,"FieldName":""} (get nodes)
2021/11/11 18:19:48 [INFO] rate limited auth-prov-v2-roletemplate-ron-2nd-do-cluster(map[objectset.rio.cattle.io/hash:bf70f8c73849ac55200b63e47052e2ee69bb0dd5]) 1.000661625s
2021/11/11 18:19:53 [INFO] rate limited auth-prov-v2-roletemplate-ron-2nd-do-cluster(map[objectset.rio.cattle.io/hash:bf70f8c73849ac55200b63e47052e2ee69bb0dd5]) 1.001928906s`

**Comment by snasovich** [comment]
https://github.com/rancher/dashboard/issues/4556 is associated UI improvement. Note that it's not a blocker to closing this issue, but just an improvement to user experience.

**Comment by ronhorton** [comment]
I'm currently testing 2 rancher docker installs, one with the feature flag off during upgrade, the other with the feature flag on. Otherwise, the setups are identical. 

first cluster (rke flag deactivated) has been configuring/provisioning for 4 hours.
second cluster B (rke flag active) has been configuring/provisioning for 25 minutes

both are `2.6-head Commit ID 638b29a`
`do - rke2 - 1 single node clusters`
`aws - rke2 - 1 single node cluster; rke1 1 single node cluster`

**Comment by ronhorton** [comment]
**PASS** Verified in `2.6-head Commit ID 78b25c3`

1. Created downstream RKE2 EC2 cluster
2. Created downstream RKE1 EC2 cluster
3. Created downstream RKE1 DO cluster
4. Created downstream RKE2 DO cluster
5. Created K3s cluster
6. attempted to deactivate RKE2 Feature Flag `Cannont deactivate RKE2 feature flag with existing RKE2 cluster(s)`
7. Successfully deleted `RKE2` clusters
8. Deactivated RKE2 feature flag `flag was deactivated as there are no RKE2 clusters`
9. Successfully deleted `RKE1` clusters

At no time were any clusters `stuck in removing state` - fixed

**Comment by ronhorton** [comment]
One use case remains for this ticket:
1. On 2.6.2 deploy  rke2 and rke1 clusters
2. Disable rke2 feature flag
3. Upgrade to 2.6head
4. Delete RKE2 cluster --> will this work?
5. Enable rke2 feature flag
6. Delete rke2 and rke1 clusters --> Will this work?

last use case is blocked by https://github.com/rancher/fleet/issues/607

**Comment by ronhorton** [comment]
Created new issue https://github.com/rancher/rancher/issues/35696 to address upgrade test.

1. On 2.6.2 deploy rke2 and rke1 clusters
2. Disable rke2 feature flag
3. Upgrade to 2.6head
4. Delete RKE2 cluster --> will this work?
5. Enable rke2 feature flag
6. Delete rke2 and rke1 clusters --> Will this work?

**Comment by ronhorton** [comment]
**PASS** Verified in `2.6-head Commit ID 4375161`
1. created rke2 do cluster
2. created rke1 do cluster
3. deleted rke2 cluster - removed from rancher, and deleted at back end
4. `disabled rke2 feature flag`
5. created 2nd rke1 do cluster
6. deleted both rke1 clusters

`All clusters were removed from rancher UI and deleted on backend`
